src/Filter/modules/SingleSelectFilter/SelectOverlay.tsx

```diff
@@ -0,0 +1,72 @@
+import { FC, useMemo } from 'react';
+import Select from 'react-select';
+import { SELECT_OVERLAY_STYLES } from './selectStyles';
+import theme from 'src/styles/theme';
+
+/**
+ * `SelectOption` is the structure of a selectable option.
+ */
+export interface SelectOption {
+    value: string | number | undefined;
+    label: string;
+}
+
+/**
+ * `SelectProps` are the props to be provided to the Select
+ * component which consists of a (hidden) standard select
+ * element and a react-select based SelectOverlay component.
+ */
+export interface SelectProps {
+    options: SelectOption[];
+    onChange(event: any): void;
+    value: string | number;
+    onBlur?(event: any): void;
```

Comment:
This might be able to be typed as React's FocusEvent<HTMLSelectElement>.

Reply:
Just made some updates. Let me know what you think.